### PR TITLE
Add download button for GitHub

### PR DESCRIPTION
**Pull request by dutiyesh**
**Add a jsDelivr CDN button for GitHub**
Show a download button if we are able to retrieve repository's name and version from it's package file

**Provide jsDelivr CDN link for a repository**
If a default file name is set for a repository, provide user with:
- A jsDelivr CDN link
- A download ZIP files button
- And a link which takes user to jsDelivr page of a repository

repo-with-default-file

---

**Show warning message if link is not available**
If CDN link is not available, add a jsDelivr page link where user can select the files.

repo-without-default-file

---

Add **mocha** and **chai** for testing the utility functions.

### CODE REVIEW

File: src/features/github-download-button.js [hunk 1]
```diff
@@ -1,3 +1,129 @@
...
+					return;
+				}
+
+				const { ownerName, repoName, treeName } = getRepoDetails();
+
+				const rawPackageUrl = `https://raw.githubusercontent.com/${ownerName}/${repoName}/${treeName}/package.json`;
+				const rawPackageResponse = await fetch(rawPackageUrl);
+				const { name, version } = await rawPackageResponse.json();
+
+				if (!name && !version) {
```
Comment (MartinKolarik) [inline src/features/github-download-button.js]: `||` ?

Comment (dutiyesh) [reply src/features/github-download-button.js]: We need both `name` and `version` for our API to get default file name from.

Comment (MartinKolarik) [reply src/features/github-download-button.js]: Sure, so if either is missing it should go to the `return`

Comment (dutiyesh) [reply src/features/github-download-button.js]: Yes.

Comment (MartinKolarik) [reply src/features/github-download-button.js]: But right now both need to be missing 😄

Comment (dutiyesh) [reply src/features/github-download-button.js]: Yes. Regarding this point, I found 1 use-case that I missed.
[Explained here](https://github.com/jsdelivr/extension-chrome/issues/1#issuecomment-537517324).

**Comment by dutiyesh** [comment]
@MartinKolarik , Updated the pull request with getting version from API.

**Pull request merged**